parsing/lexer.mll

```diff
@@ -157,15 +157,20 @@ let char_for_backslash = function
   | 't' -> '\009'
   | c   -> c
 
+let illegal_escape lexbuf reason =
+  let error = Illegal_escape (Lexing.lexeme lexbuf, Some reason) in
+  raise (Error (error, Location.curr lexbuf))
+
 let char_for_decimal_code lexbuf i =
   let c = 100 * (Char.code(Lexing.lexeme_char lexbuf i) - 48) +
            10 * (Char.code(Lexing.lexeme_char lexbuf (i+1)) - 48) +
                 (Char.code(Lexing.lexeme_char lexbuf (i+2)) - 48) in
   if (c < 0 || c > 255) then
     if in_comment ()
     then 'x'
-    else raise (Error(Illegal_escape (Lexing.lexeme lexbuf),
-                      Location.curr lexbuf))
+    else
+      illegal_escape lexbuf
+        (Printf.sprintf "%d is outside the range of ASCII characters." c)
```

Comment:
Small note: technically, ASCII chars are limited to 0-127, while this flags things above 255. I think the error message

`%d is outside the range of legal characters (0-255)`

might be slightly better.

Reply:
Oh, and thanks for this change, it is a significantly better way to notify the user of the error condition.